jaseci_core/jaseci/svc/common.py

```diff
@@ -288,19 +290,53 @@ def __init__(self, meta):
             actions_history=self.actions_history,
             actions_calls=self.actions_calls,
         )
+        self.db_check()
+
+    def db_check(self):
+        try:
+            if db["enabled"]:
+                connection = psycopg2.connect(
+                    host=db["host"],
+                    dbname=db["db"],
+                    user=db["user"],
+                    password=db["password"],
+                    port=db["port"],
+                )
+                connection.close()
+            self.has_db = True
+        except Exception:
+            self.has_db = False
+
+    def db_regen(self):
+        for kind, confs in POSTGRES_MANIFEST.items():
+            for conf in confs:
+                name = conf["metadata"]["name"]
+                res = self.read(kind, name, self.namespace)
+                if hasattr(res, "status") and res.status == 404 and conf:
+                    self.create(kind, name, self.namespace, conf)
+
+        self.db_check()
+        if self.has_db:
+            for item in self.kubernetes.core.list_namespaced_pod(
+                namespace=self.namespace, label_selector=f"pod={db['pod']}"
+            ).items:
+                self.kubernetes.core.delete_namespaced_pod(item["name"], self.namespace)
 
     ###################################################
     #                     BUILDER                     #
     ###################################################
 
     def build(self):
         try:
-            hook = self.build_context("hook")
-            config = hook.service_glob("META_CONFIG", META_CONFIG)
+            config = META_CONFIG
+            if self.has_db:
+                hook = self.build_context("hook")
+                config = hook.service_glob("META_CONFIG", META_CONFIG)
+                self.prometheus = self.meta.get_service("promon", hook)
```

Comment:
Why is prometheus hoisted up here?

Reply:
Prometheus is part of JsOrc and pointed to JsOrc's Kubernetes so I just initialized it when JsOrc is automated